lib/src/Uncertainty/Algorithm/Optimization/EfficientGlobalOptimization.cxx

```diff
@@ -94,7 +96,7 @@ class ExpectedImprovementEvaluation : public EvaluationImplementation
   Point operator()(const Point & x) const
   {
     const Scalar mx = metaModelResult_.getConditionalMean(x)[0];
-    const Scalar fmMk = optimalValue_ - mx;
+    const Scalar fmMk = isMinimization_ ? optimalValue_ - mx : mx - optimalValue_;
     const Scalar sk2 = metaModelResult_.getConditionalMarginalVariance(x);
     const Scalar sk = sqrt(sk2);
     if (!SpecFunc::IsNormal(sk)) return Point(1, SpecFunc::LowestScalar);
```

Comment:
```suggestion
    if (!SpecFunc::IsNormal(sk))
    {
      if (isMinimization_) return Point(1, SpecFunc::LowestScalar);
      else return Point(1, SpecFunc::MaxScalar);
    }
```

Reply:
This flag is about the global objective
I think the criterion is always maximized regardless if the global objective is minimized or maximized isnt  it ?

Reply:
you are right!